common/flatpak-installation.c

```diff
@@ -1440,6 +1440,27 @@ flatpak_installation_get_config (FlatpakInstallation *self,
   return flatpak_dir_get_config (dir, key, error);
 }
 
+/**
+ * flatpak_installation_get_min_free_space:
+ * @self: a #FlatpakInstallation
+ *
+ * A convenient wrapper around ostree_repo_get_min_free_space.
+ *
+ * Returns: Value (in bytes) of min-free-space-* config option of
+ * the Ostree repo.
+ */
+guint64
+flatpak_installation_get_min_free_space_bytes (FlatpakInstallation *self)
+{
+  g_autoptr(FlatpakDir) dir = NULL;
+
+  dir = flatpak_installation_get_dir (self, NULL);
+  if (dir == NULL)
+    return 0;
+
+  return ostree_repo_get_min_free_space_bytes (flatpak_dir_get_repo (dir));
```

Comment:
This will cause an assertion failure if the repository is not in the middle of a transaction. How do you intend to tie that into the flatpak API, so people can call this function at a sensible time and _not_ get an assertion failure?

To answer that question, it would be helpful to know where you plan to call this function from, and to iterate the design in parallel with developing whatever branch is making the call.

Reply:
Well. this is a very solid point and I am very sure I already have made a goof-up.

This was intended to replace `EOS_MIN_FREE_SPACE_NEEDED_BYTES` which is basically a hard-coded `min-free-space`(`500MB`) for the install/update success heuristics in gnome-software I recently developed:  https://github.com/endlessm/gnome-software/pull/399/files#diff-e64404ac7c111b350b85abaa4fb8084dR59